test-apps/ember-concurrency-v2/tests/unit/wait-for-test.ts

```diff
@@ -0,0 +1,442 @@
+import MockStableError, {
+  overrideError,
+  resetError,
+} from './utils/mock-stable-error';
+// @ember/test-waiters is still a v1 addon and is too weird
+// to have in-repo types working correctly.
+// @ts-ignore
+import { _reset, getPendingWaiterState, waitFor } from '@ember/test-waiters';
+import { module, test } from 'qunit';
+
+import EmberObject, { get } from '@ember/object';
+import { DEBUG } from '@glimmer/env';
+
+import { task as taskFn, TaskGenerator, didCancel } from 'ember-concurrency';
+// type resolution is not working correctly due to usage of forked
+// (non-published) ember-concurrency-decorators remove this ts-ignore when
+// migrating back to mainline off the fork
+// @ts-ignore
+import { task as taskDec } from 'ember-concurrency-decorators';
+import { perform } from 'ember-concurrency-ts';
+// @ts-ignore
+import co from 'co';
+
+// @ember/test-waiters is still a v1 addon and is too weird
+// to have in-repo types working correctly.
+// @ts-ignore
```

Comment:
ts-ignores can be cleaned up once test-waiters is a v2 addon and actually has types we can consume